### PR TITLE
Fix issues found by lgtm

### PR DESCRIPTION
**Pull request by xiemaisi**
This fixes a few minor JavaScript issues found by [lgtm](https://lgtm.com); for more details see https://lgtm.com/projects/g/uber/deck.gl.

Each commit fixes one particular kind of issue; I'd be happy to squash them into one if preferred.

(Full disclosure: I work on lgtm's JavaScript analysis.)

### CODE REVIEW

File: src/core/utils/bin-sorter.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default class BinSorter {
       .reduce((accu, h, i) => {
         const value = getValue(h.points);
 
-        if (value !== null || value !== undefined) {
+        if (value !== null && value !== undefined) {
```
Comment (ibgreen) [inline src/core/utils/bin-sorter.js]: Looks right to me. @heshan0131 fyi

**Comment by ibgreen** [comment]
> Each commit fixes one particular kind of issue; I'd be happy to squash them into one if preferred.

No problems. We squash when merging.

> (Full disclosure: I work on lgtm's JavaScript analysis.)

Very nice! It seems to be working well. Some of the findings are actually minor bugs. Maybe doing an occasional lgtm run is something we should consider for more of our repos, perhaps our main frameworks (see vis.gl).

**Comment by xiemaisi** [comment]
> Very nice! It seems to be working well.

Thanks, I'm glad to hear you find the results useful. The unused dependency warnings seem to be false positives, I'll look into improving that rule.

> Maybe doing an occasional lgtm run is something we should consider for more of our repos

You might find our [CI integration](https://lgtm.com/docs/lgtm/using-lgtm-analysis-continuous-integration) useful: once you turn it on, every GitHub pull request is analysed and information about new and fixed alerts is displayed as part of the PR status. Here is an example of what it looks like in practice: https://github.com/Open-MBEE/MDK/pull/132#issuecomment-331294008

**Comment by ibgreen** [comment]
@balthazar @KevinGrandon About LGTM integration ^

**Comment by balthazar** [comment]
That seem pretty nice, @ibgreen regarding all the alerts concerning the extraneous luma dependency, should we remove it from the examples dependencies since it's already a hard dep of deck, or do you think there would be demand for luma being a peer of deck?

**Comment by ibgreen** [comment]
We could remove it, at least for apps that don't directly import luma in their source code (i.e. they only use luma through deck.gl).

**Comment by xiemaisi** [comment]
@ibgreen / @balthazar, any thoughts on enabling pull request integration? It's a relatively new feature of lgtm, and we are keen on getting people to try it out and tell us how we can make it more useful!

**Comment by balthazar** [comment]
Hey @xiemaisi! Yes I can try it out on another smaller project and see how that goes before doing it here, I'm concerned having too many integrations might bloat the PRs threads with bot replies 😛

**Comment by xiemaisi** [comment]
If you're worried about bot spam, you can choose to disable PR comments when setting up the integration; you can then get a summary of the analysis results by hovering over the build status, but no comments will be posted. (Even if you do enable comments, you'll only get one comment per analysis, not one per alert, so it's generally still pretty manageable.)

**Pull request merged**